moby/moby
# depend on docker-selinux for fedora-25+ and centos-7

**- What I did**

Changed the dependency in the docker-engine RPM built for Fedora 25+ to depend on `docker-selinux` which is satisfied by the `container-selinux` package. This PR fixes issue #28099.

Also, now that [centos 7.3 has been released](https://lists.centos.org/pipermail/centos-announce/2016-December/022172.html), changed the dependency in the `docker-engine` RPM built for CentOS to depend on `docker-selinux` which happens to be `docker-selinux-1.10.3-46.el7.centos.14.x86_64` but should still work for docker-engine all the way up through 1.13.

**- How I did it**

Added a conditional `Requires` in the `docker-engine.spec` file for fedora-25.

Added a section to the `docker-engine.spec` file for defining a subpackage. This will cause centos-7 builds to create a `docker-engine-selinux` [subpackage](http://rpm.org/max-rpm-snapshot/ch-rpm-subpack.html) (which is just another RPM built from the same spec file) that has no files but just a dependency on the `docker-selinux` package. 

Also refactored the `hack/make/build-rpm` to only build a `docker-engine-selinux` package if there exists a specific selinux policy directory in the `contrib` dir. In doing so, found that the `contrib/selinux` policy dir is no longer used by anybody so deleted it.

Note: this changes the default behaviour of building RPMs from using a generic selinux policy to only using one for the specificy distro version.

**- How to verify it**

Build the fedora-25 rpm and install on a Fedora 25 machine with selinux enabled.

```
$ make DOCKER_BUILD_PKGS=fedora-25 rpm
```

Build a `docker-engine` centos-7 RPM package and install on centos 7.3 with selinux enabled:
```
$ make DOCKER_BUILD_PKGS=centos-7 rpm
```

Configure docker daemon to run with selinux by setting `/etc/docker/daemon.json`:
```json
{
  "selinux-enabled": true
}
```

See that the `svirt_sandbox_file_t` label is set which means docker is doing the right thing with selinux enabled:
```
$ sudo docker run --rm ubuntu ls -alZ /home
total 4
drwxr-xr-x.  2 root root system_u:object_r:svirt_sandbox_file_t:s0:c407,c628    6 Apr 12  2016 .
drwxr-xr-x. 21 root root system_u:object_r:svirt_sandbox_file_t:s0:c407,c628 4096 Dec 14 04:49 ..
```

**- Description for the changelog**


Use the selinux policy provided by `docker-selinux` on Fedora 25+ and CentOS 7.3.

**- A picture of a cute animal (not mandatory but encouraged)**

🐺
